### ISSUE TITLE
`>=` typo gives unclear error message

### ISSUE THREAD

**Issue by macmv** [rust-lang/rust#98128]
Given the following code: 

```rust
let is_greater = 5 => 3;
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: expected one of `.`, `;`, `?`, `else`, or an operator, found `=>`
 --> src/main.rs:2:24
  |
2 |     let is_greater = 5 => 3;
  |                        ^^ expected one of `.`, `;`, `?`, `else`, or an operator

error: could not compile `playground` due to previous error
```
[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=524ff9f12a75ace09945812df9c9c3b9)

This had me very confused for a surprising amount of time. It would be nice if the compiler would give a suggestion to replace `=>` with `>=`. In my opinion, this would help a lot for someone typing quickly, and would make it clear that the symbols were in reverse. It might also lead to some false positives, but I think it would still be nice to have.

The error is actually worse for macros:

```rust
macro_rules! accept_expr {
    ( $expr:expr ) => { $expr }
}

fn main() {
    accept_expr!(5 => 3);
}
```

The current output for this macro is:

```
   Compiling playground v0.0.1 (/playground)
error: no rules expected the token `=>`
 --> src/main.rs:6:20
  |
1 | macro_rules! accept_expr {
  | ------------------------ when calling this macro
...
6 |     accept_expr!(5 => 3);
  |                    ^^ no rules expected this token in macro call

error: could not compile `playground` due to previous error
```

My suggestion it to simply add a `help` section to both of the above errors, which would suggest replacing the `=>` token with `>=`.

I'm interested in fixing this. Where should I start? I have `rustc` compiled, but I don't really know where this error is even produced.

**Comment by compiler-errors** [comment]
This is complicated, given that `=>` is parsed as its own token as opposed the equivalent typo `=<` for the lt operator. I'm not confident that fixing this (at least in the `macro_rules!` case, but possibly in arbitrary positions) is possible, because this is a valid macro:

```
macro_rules! foo {
    ($a:expr => $b:expr) => {};
}
```

**Comment by ChayimFriedman2** [comment]
If this is possible, it would also be nice to mention closures there, since `a => b` is the closure syntax in many languages.

**Comment by veera-sivarajan** [comment]
Output on `rustc 1.79.0`:
```
error: expected one of `.`, `;`, `?`, `else`, or an operator, found `=>`
 --> <source>:3:24
  |
3 |     let is_greater = 5 => 3;
  |                        ^^ expected one of `.`, `;`, `?`, `else`, or an operator
  |
help: you might have meant to write a "greater than or equal to" comparison
  |
3 |     let is_greater = 5 >= 3;
  |                        ~~

error: aborting due to 1 previous error

```
